app/models/account.rb

```diff
@@ -36,18 +39,54 @@ class Account < ApplicationRecord
   # @return [Account] A persisted instance of Account
   #
   def self.cache_or_create_by!(user)
-    return unless user.uuid
+    return unless user.uuid || user.sec_id
 
-    do_cached_with(key: user.uuid) do
+    # if possible use the idme uuid for the key, fallback to using the sec id otherwise
+    key = user.uuid || "sec:#{user.sec_id}"
+    acct = do_cached_with(key: key) do
       create_if_needed!(user)
     end
+    # Account.sec_id was added months after this class was built, thus
+    # the existing Account records (not new ones) need to have their
+    # sec_id value updated
+    update_if_needed!(acct, user)
   end
 
   def self.create_if_needed!(user)
-    find_or_create_by!(idme_uuid: user.uuid) do |account|
-      account.edipi = user&.edipi
-      account.icn   = user&.icn
+    attrs = account_attrs_from_user(user)
+
+    accts = where(idme_uuid: attrs[:idme_uuid])
+            .where.not(idme_uuid: nil)
+            .or(
+              where(sec_id: attrs[:sec_id])
+              .where.not(sec_id: nil)
+            )
+    if accts.length > 1
+      # TODO: are any ids in an Account record considered PII? if so we need
+      # to change the extra_context value
+      log_message_to_sentry('multiple Account records with matching ids', 'warning', accts)
     end
+
+    accts.length.positive? ? accts[0] : create(**attrs)
+  end
+
+  def self.update_if_needed!(account, user)
+    # account has yet to be saved, no need to update
+    return account unless account.id
+
+    # return account as is if all user attributes match up to be the same
+    attrs = account_attrs_from_user(user)
+    return account if attrs.all? { |k, v| account.send(k) == v }
+
```

Comment:
I would like to consider logging the old and new values of account/attrs here. Basically if we have a logic error or some misunderstanding of the constraints around identifiers in some external system, we potentially want to be able to reconstruct the updates that we are making to these records. 
If so we should call out in PR review that we are logging VA-internal identifiers to cloudwatch, which is allowed per ATO but should still be a conscious decision.

Reply:
will do, in general what critiera do you use to decided if the log should go to sentry or cloudwatch?